<ui version="4.0" >
 <class>PathFinderBase</class>
 <widget class="QDialog" name="PathFinderBase" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>410</width>
    <height>364</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Boolean Path Operations</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2" >
   <item>
    <layout class="QHBoxLayout" >
     <property name="spacing" >
      <number>5</number>
     </property>
     <item>
      <layout class="QVBoxLayout" >
       <property name="spacing" >
        <number>5</number>
       </property>
       <item>
        <widget class="QLabel" name="sourceShape" >
         <property name="minimumSize" >
          <size>
           <width>100</width>
           <height>100</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>100</width>
           <height>100</height>
          </size>
         </property>
         <property name="toolTip" >
          <string>First source shape.</string>
         </property>
         <property name="autoFillBackground" >
          <bool>true</bool>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Panel</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="text" >
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="keepSource1" >
         <property name="toolTip" >
          <string>Keep a copy of the original item after applying the operation</string>
         </property>
         <property name="text" >
          <string>Keep</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" >
       <property name="spacing" >
        <number>0</number>
       </property>
       <item>
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>2</width>
           <height>2</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="label" >
         <property name="font" >
          <font>
           <pointsize>30</pointsize>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text" >
          <string>+</string>
         </property>
         <property name="alignment" >
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>2</width>
           <height>2</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" >
       <property name="spacing" >
        <number>5</number>
       </property>
       <item>
        <widget class="QLabel" name="sourceShape2" >
         <property name="minimumSize" >
          <size>
           <width>100</width>
           <height>100</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>100</width>
           <height>100</height>
          </size>
         </property>
         <property name="toolTip" >
          <string>Second source shape.</string>
         </property>
         <property name="autoFillBackground" >
          <bool>true</bool>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Panel</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="text" >
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="keepSource2" >
         <property name="toolTip" >
          <string>Keep a copy of the original item after applying the operation</string>
         </property>
         <property name="text" >
          <string>Keep</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" >
       <property name="spacing" >
        <number>0</number>
       </property>
       <item>
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>2</width>
           <height>2</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="label_2" >
         <property name="font" >
          <font>
           <pointsize>30</pointsize>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text" >
          <string>=</string>
         </property>
         <property name="alignment" >
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>2</width>
           <height>2</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" >
       <item>
        <widget class="QLabel" name="resultShape" >
         <property name="minimumSize" >
          <size>
           <width>100</width>
           <height>100</height>
          </size>
         </property>
         <property name="maximumSize" >
          <size>
           <width>100</width>
           <height>100</height>
          </size>
         </property>
         <property name="toolTip" >
          <string>The resulting shape.</string>
         </property>
         <property name="autoFillBackground" >
          <bool>true</bool>
         </property>
         <property name="frameShape" >
          <enum>QFrame::Panel</enum>
         </property>
         <property name="frameShadow" >
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="text" >
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout" >
     <item>
      <layout class="QVBoxLayout" name="verticalLayout" >
       <item>
        <widget class="QGroupBox" name="groupBox" >
         <property name="title" >
          <string>Operation</string>
         </property>
         <layout class="QHBoxLayout" >
          <property name="spacing" >
           <number>5</number>
          </property>
          <property name="margin" >
           <number>10</number>
          </property>
          <item>
           <widget class="QToolButton" name="opCombine" >
            <property name="toolTip" >
             <string>Unites the shapes</string>
            </property>
            <property name="text" >
             <string>...</string>
            </property>
            <property name="checkable" >
             <bool>true</bool>
            </property>
            <property name="checked" >
             <bool>true</bool>
            </property>
            <property name="autoExclusive" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="opSubtraction" >
            <property name="toolTip" >
             <string>Subtracts the second shape from the first shape</string>
            </property>
            <property name="text" >
             <string>...</string>
            </property>
            <property name="checkable" >
             <bool>true</bool>
            </property>
            <property name="autoExclusive" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="opIntersection" >
            <property name="toolTip" >
             <string>Intersection of the shapes</string>
            </property>
            <property name="text" >
             <string>...</string>
            </property>
            <property name="checkable" >
             <bool>true</bool>
            </property>
            <property name="autoExclusive" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="opExclusion" >
            <property name="toolTip" >
             <string>Result is the area where the two shapes do not intersect</string>
            </property>
            <property name="text" >
             <string>...</string>
            </property>
            <property name="checkable" >
             <bool>true</bool>
            </property>
            <property name="autoExclusive" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QToolButton" name="opParts" >
            <property name="toolTip" >
             <string>Break apart,
The result is a combination of "Intersection" and "Exclusion"</string>
            </property>
            <property name="text" >
             <string>...</string>
            </property>
            <property name="checkable" >
             <bool>true</bool>
            </property>
            <property name="autoExclusive" >
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" >
         <property name="spacing" >
          <number>0</number>
         </property>
         <item>
          <widget class="QPushButton" name="swapShapes" >
           <property name="toolTip" >
            <string>Exchange the Source Shapes</string>
           </property>
           <property name="text" >
            <string>Swap Shapes</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer>
           <property name="orientation" >
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0" >
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer" >
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="optionGroup" >
       <property name="title" >
        <string>Result takes color from:</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2" >
        <item row="0" column="0" >
         <widget class="QRadioButton" name="targetGetsSource1Color" >
          <property name="toolTip" >
           <string>The resulting shape uses the color of the first source shape</string>
          </property>
          <property name="text" >
           <string>First Shape</string>
          </property>
          <property name="checked" >
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0" >
         <widget class="QRadioButton" name="targetGetsSource2Color" >
          <property name="toolTip" >
           <string>The resulting shape uses the color of the second source shape</string>
          </property>
          <property name="text" >
           <string>Second Shape</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0" >
         <widget class="QRadioButton" name="targetGetsOtherColor" >
          <property name="toolTip" >
           <string>The resulting shape uses the colors listed below</string>
          </property>
          <property name="text" >
           <string>Custom Colors</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0" >
         <layout class="QGridLayout" name="gridLayout" >
          <item row="0" column="0" >
           <widget class="QLabel" name="label_3" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="text" >
             <string>Stroke:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" >
           <widget class="ColorCombo" name="otherColorComboLine" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="sizePolicy" >
             <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="toolTip" >
             <string>Stroke Color</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" >
           <widget class="QLabel" name="label_4" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="text" >
             <string>Fill:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" >
           <widget class="ColorCombo" name="otherColorComboFill" >
            <property name="enabled" >
             <bool>false</bool>
            </property>
            <property name="toolTip" >
             <string>Fill Color</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox" >
     <property name="orientation" >
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons" >
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="5" margin="10" />
 <customwidgets>
  <customwidget>
   <class>ColorCombo</class>
   <extends>QComboBox</extends>
   <header>colorcombo.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PathFinderBase</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>307</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel" >
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PathFinderBase</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>375</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel" >
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsSource1Color</sender>
   <signal>clicked(bool)</signal>
   <receiver>otherColorComboLine</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>388</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel" >
     <x>387</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsSource1Color</sender>
   <signal>clicked(bool)</signal>
   <receiver>otherColorComboFill</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>388</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel" >
     <x>387</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsSource2Color</sender>
   <signal>clicked(bool)</signal>
   <receiver>otherColorComboLine</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>388</x>
     <y>215</y>
    </hint>
    <hint type="destinationlabel" >
     <x>387</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsSource2Color</sender>
   <signal>clicked(bool)</signal>
   <receiver>otherColorComboFill</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>388</x>
     <y>215</y>
    </hint>
    <hint type="destinationlabel" >
     <x>387</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsOtherColor</sender>
   <signal>clicked(bool)</signal>
   <receiver>otherColorComboLine</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>388</x>
     <y>238</y>
    </hint>
    <hint type="destinationlabel" >
     <x>387</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsOtherColor</sender>
   <signal>clicked(bool)</signal>
   <receiver>otherColorComboFill</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>388</x>
     <y>238</y>
    </hint>
    <hint type="destinationlabel" >
     <x>387</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsSource1Color</sender>
   <signal>clicked(bool)</signal>
   <receiver>label_3</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>262</x>
     <y>182</y>
    </hint>
    <hint type="destinationlabel" >
     <x>245</x>
     <y>257</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsSource2Color</sender>
   <signal>clicked(bool)</signal>
   <receiver>label_3</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>282</x>
     <y>211</y>
    </hint>
    <hint type="destinationlabel" >
     <x>260</x>
     <y>253</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsOtherColor</sender>
   <signal>clicked(bool)</signal>
   <receiver>label_3</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>273</x>
     <y>225</y>
    </hint>
    <hint type="destinationlabel" >
     <x>264</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsSource1Color</sender>
   <signal>clicked(bool)</signal>
   <receiver>label_4</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>319</x>
     <y>185</y>
    </hint>
    <hint type="destinationlabel" >
     <x>243</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsSource2Color</sender>
   <signal>clicked(bool)</signal>
   <receiver>label_4</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>298</x>
     <y>207</y>
    </hint>
    <hint type="destinationlabel" >
     <x>259</x>
     <y>284</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetGetsOtherColor</sender>
   <signal>clicked(bool)</signal>
   <receiver>label_4</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>257</x>
     <y>234</y>
    </hint>
    <hint type="destinationlabel" >
     <x>256</x>
     <y>286</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
